server/users.js

```diff
@@ -1,4 +1,6 @@
-const users = [];
+const users = {};
```

Comment:
When a user clicks on start, they are already both waiting and active users tho

Reply:
When a user clicks on start it the server will get to know one user has joined and it will take this user to waiting area. Now there will be one background process will be running to check every 3 sec whether the number of users in waiting list is greater than 1 or not if greater then 1 then we do the further process i.e creating room adding it to active user and user list and atlast one message will be send to front-end to make isfound true then user will be easily able to chat.

Reply:
> When a user clicks on start it the server will get to know one user has joined and it will take this user to waiting area. Now there will be one background process will be running to check every 3 sec whether the number of users in waiting list is greater than 1 or not if greater then 1 then we do the further process i.e creating room adding it to active user and user list and atlast one message will be send to front-end to make isfound true then user will be easily able to chat.

i know, i'm just letting you know that i already made it so when a user clicks on start they are already waiting, pressing start puts every user in waiting, so i don't see the use for waiting area

Reply:
OK.